src/components/calculator/PrevalenceControls.tsx

```diff
@@ -306,31 +330,28 @@ export const PrevalenceControls: React.FunctionComponent<{
           />
         </div>
       )}
-      {isManualEntryCurrently ? (
-        <span>
-          <button
-            id="switchBetweenManualDataAndLocationSelection"
-            className="btn btn-link text-muted"
-            onClick={handleSelectLocationButtonOnClick}
-          >
-            {t('calculator.switch_button.select_location')}
-          </button>
-        </span>
-      ) : (
-        <span>
-          <button
-            id="switchBetweenManualDataAndLocationSelection"
-            className="btn btn-link text-muted"
-            onClick={handleEnterDataButtonOnClick}
-          >
+      <span>
+        <ToggleButton
+          id="switchBetweenManualDataAndLocationSelection"
+          name={t('calculator.switch_button.select_location')}
+          type="checkbox"
+          checked={isManualEntryCurrently}
+          value="1"
+          variant="link"
+          className="text-muted"
+          onChange={(e: React.ChangeEvent<HTMLInputElement>) => {
+            setManualPrevalenceData(e.currentTarget.checked)
+          }}
+        >
+          <span id="switchBetweenManualDataAndLocationSelectionText">
             {t('calculator.switch_button.enter_data_manually')}
-          </button>
-        </span>
-      )}
+          </span>
+        </ToggleButton>
+      </span>
       <ControlledExpandable
         id="prevalence-details"
         header={t('calculator.prevalence.details_header')}
-        headerClassName={isManualEntryCurrently ? 'd-none' : ''}
+        noContract={isManualEntryCurrently}
```

Comment:
Is this field being added because the manual entry is actually inside the expandable?

That seems odd to me. I think this would be easier to reason about this if a different component that gets swapped in rather than having a complicated override of how an expandable works.

Reply:
the manual entry swap was done at the PrevalenceField level. If isManualEntryCurrently was true, it would flip from the form input to the "statement" form. The tradeoff is a bit more mess in ControlledExpandable to get the smooth animation working between manual entry and details. Getting that animation working properly without having it all be in the same <Collapse> is really tricky.

See the WIP commit, which cleans up ControlledExpandable (and some of PrevalenceControls) but removes the animation between manual and location mode.